classes/Tools.php

```diff
@@ -627,7 +627,7 @@ public static function setCurrency($cookie)
             }
         }
 
-        return $currency;
+        return empty($currency) ? new Currency() : $currency;
```

Comment:
Not the best solution in my opinion.

The real issue is line 623, where we use the same variable name for another purpose, that's why we get trouble on the reported ticket.

If we reach the line you modified, you will _always_ have an array. And the `empty($currency)` won't fix the issue if some data is returned from the database.

If this function is supposed to return the default currency, then reuse the example you got a few lines above

```php
return Currency::getCurrencyInstance(Configuration::get('PS_CURRENCY_DEFAULT'));
```